cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -3321,4 +3322,57 @@ private long getMetric(ApplicationId appId, String metric) {
                                                Constants.Metrics.Tag.WORKFLOW, SmartWorkflow.NAME);
     return getMetricsManager().getTotalMetric(tags, "user." + metric);
   }
+
+  @Test
+  public void testConfigProperties() throws Exception {
```

Comment:
this test should not be in the data pipeline module. It is a platform feature so the test should be in the platform. You can test this with a MapReduce program where the input format and output format set different values to the same key.